packages/alchemy/src/middleware/gas-manager.ts

```diff
@@ -112,13 +112,16 @@ const withAlchemyGasAndPaymasterAndDataMiddleware = (
       await resolveProperties(struct)
     );
 
-    let feeOverride = undefined;
-    if (userOperation.maxFeePerGas && BigInt(userOperation.maxFeePerGas) > 0n) {
-      feeOverride = {
-        maxFeePerGas: userOperation.maxFeePerGas,
-        maxPriorityFeePerGas: userOperation.maxPriorityFeePerGas,
-      };
-    }
+    const overrides =
+      userOperation.maxFeePerGas && BigInt(userOperation.maxFeePerGas) > 0n
+        ? {
+            maxFeePerGas: userOperation.maxFeePerGas,
+            maxPriorityFeePerGas: userOperation.maxPriorityFeePerGas,
+            callGasLimit: userOperation.callGasLimit,
+            verificationGasLimit: userOperation.verificationGasLimit,
+            prevVerificationGas: userOperation.preVerificationGas,
+          }
+        : undefined;
```

Comment:
if in the above, it's possible to override individual items, then we need to check each field individually to make sure they're greater > 0 because the alchemy gas manager middleware sets all the gas + fee fields to 0.

If someone overrides just the fee fields, then we'd be setting the gas limits to 0 and the request will fail